src/gamma.jl

```diff
@@ -398,6 +398,49 @@ function _invdigamma(y::Float64)
     return x_new
 end
 
+"""
+    invtrigamma(x)
+Compute the inverse [`trigamma`](@ref) function of `x`.
+"""
+invtrigamma(y::Number) = _invtrigamma(float(y))
+
+function _invtrigamma(y::Float64)
+    # Implementation of Newton algorithm described in
+    # "Linear Models and Empirical Bayes Methods for Assessing
+    #  Differential Expression in Microarray Experiments"
+    # (Appendix "Inversion of Trigamma Function")
+    #  by Gordon K. Smyth, 2004
+
+    if y <= 0
+        throw(DomainError(y, "Only positive `y` supported."))
+    end
+
+    if y > 1e7
+        return inv(sqrt(y))
+    elseif y < 1e-6
+        return inv(y)
+    end
+
+    x_old = inv(y) + 0.5
+    x_new = x_old
+
+    # Newton iteration
+    δ = Inf
+    iteration = 0
+    while δ > 1e-8 && iteration <= 25
+        iteration += 1
+        f_x_old = trigamma(x_old)
+        δx =  f_x_old*(1-f_x_old/y) / polygamma(2, x_old)
+        x_new = x_old + δx
+        δ = - δx / x_new
+        x_old = x_new
+    end
+
+    return x_new
```

Comment:
```suggestion
    # Newton iteration
    invx = inv(x)
    y_new = y_old = invx + 0.5
    for _ in 1:26
        ψ′ = trigamma(y_old)
        δ = ψ′ * (1 - ψ′ * invx) / polygamma(2, y_old)
        y_new = y_old + δ
        -δ / y_new < 1e-8 && break
        y_old = y_new
    end
    return y_new
```
AFAICT this is equivalent and more directly maps to the paper, as it avoids introducing a second step size variable. You can also avoid dividing by the input at every iteration by inverting once then multiplying by the inverse.

I assume the number of iterations was chosen to match `invdigamma`. Do you know whether the algorithm generally converges within the given number of iterations and under what circumstances it may not? When it doesn't, how inaccurate is the result?